### PR TITLE
Make ServeHTTP responses Vary on Accept

### PR DESCRIPTION
**Pull request by ithinkihaveacat**
I think this is the right place to do it?

Ideally should also check that non-SXG responses also `Vary: Accept` but not sure where to do this.

### CODE REVIEW

File: packager/signer/signer.go [hunk 1]
```diff
@@ -179,6 +179,8 @@ func (this *Signer) genCertURL(cert *x509.Certificate, signURL *url.URL) (*url.U
 }
 
 func (this *Signer) ServeHTTP(resp http.ResponseWriter, req *http.Request, params httprouter.Params) {
+	resp.Header().Add("Vary", "Accept")
```
Comment (twifkak) [inline packager/signer/signer.go]: While you're here, make it "Accept, AMP-Cache-Transform".

Comment (twifkak) [reply packager/signer/signer.go]: I'm slightly partial to a single header `Vary: Accept, AMP-Cache-Transform` as it's slightly simpler (intermediaries are supposed to handle duplicate header names, but I wouldn't be surprised if some don't).

File: packager/signer/signer_test.go [hunk 2]
```diff
@@ -155,6 +155,7 @@ func (this *SignerSuite) TestSimple() {
 	this.Assert().Equal(fmt.Sprintf(`google;v="%d"`, transformer.SupportedVersions[0].Max), resp.Header.Get("AMP-Cache-Transform"))
 	this.Assert().Equal("nosniff", resp.Header.Get("X-Content-Type-Options"))
 	this.Assert().Equal(fakePath, this.lastRequestURL)
+	this.Assert().Equal("Accept, AMP-Cache-Transform", resp.HeaderGet("Vary")
```
Comment (twifkak) [inline packager/signer/signer_test.go]: Missing end paren. :)

Comment (ithinkihaveacat) [reply packager/signer/signer_test.go]: Sigh. Fixed now!

**Comment by twifkak** [comment]
Thanks, @ithinkihaveacat! Travis is happy, so I'm merging.

**Pull request merged**